### PR TITLE
Logging improvements

### PR DESCRIPTION
**Pull request by ratoaq2**
- Fixing duplicate logging message (when enabling subliminal logging)
- Moving threadName and 'current commit hash' to the logger formatter in order to have the same message pattern for modules other than sickbeard.

Here's a preview:

```
2016-04-05 21:40:15 INFO     Thread-16 :: [6bb3888] No subtitles found for /library/series/Real Humans/Season 02/Real Humans - S02E10 - Episode 10.mkv
2016-04-05 21:40:15 INFO     Thread-16 :: [6bb3888] Searching candidates using the keyword real humans
2016-04-05 21:40:14 INFO     Thread-16 :: [6bb3888] Logging in
2016-04-05 21:40:14 INFO     Thread-16 :: [6bb3888] Initializing provider legendastv
2016-04-05 21:40:14 INFO     Thread-16 :: [6bb3888] Listing subtitles with provider 'legendastv' and languages set([])
2016-04-05 21:40:14 INFO     Thread-16 :: [6bb3888] Refining video with omdb
2016-04-05 21:40:13 INFO     Thread-16 :: [6bb3888] Getting series episode 2x10
2016-04-05 21:40:12 INFO     Thread-16 :: [6bb3888] Searching series 'Real Humans'
2016-04-05 21:40:12 INFO     Thread-16 :: [6bb3888] Refining video with tvdb
2016-04-05 21:40:12 INFO     Thread-16 :: [6bb3888] Refining video with release
2016-04-05 21:40:12 INFO     Thread-16 :: [6bb3888] Refining video with metadata
2016-04-05 21:40:12 INFO     Thread-16 :: [6bb3888] Scanning video 'Real Humans - S02E10 - Episode 10.mkv' in '/library/series/Real Humans/Season 02'
2016-04-05 21:39:38 INFO     TORNADO :: [6bb3888] Starting Medusa on http://0.0.0.0:8081/
```

Additional configuration could be later added to enable/disable certain modules to be logged.

### I'm still testing to see if everything is working as expected.

### CODE REVIEW

**Comment by fernandog** [comment]
@medariox

**Comment by ratoaq2** [comment]
I would like to have an option to enable/disable logging for each module: subliminal, tornado, etc. And/Or to configure different level for each module.

When I have free time I'll make sure the errors/warnings from those modules that uses standard python logging api are submitable. The good side effect is that even medusa could use the standard python api and have the same features that exist today: filter,  search, submit.

File: sickbeard/logger.py [hunk 1]
```diff
@@ -211,7 +233,7 @@ def log(self, msg, level=INFO, *args, **kwargs):
             r'error \[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE\] sslv3 alert handshake failure \(_ssl\.c:\d+\)',
         ]
         for ssl_error in ssl_errors:
-            check = re.sub(ssl_error, 'See: http://git.io/vuU5V', message)
+            check = re.sub(ssl_error, 'See: http://git.io/vVaIj', message)
```
Comment (OmgImAlexis) [inline sickbeard/logger.py]: This can be changed to use `https`.

Comment (ratoaq2) [inline sickbeard/logger.py]: Sure. I do agree on it.

**Comment by fernandog** [comment]
rebased

**Comment by fernandog** [comment]
working fine the logFilter and the logSearch

**Comment by ratoaq2** [comment]
Now I now how sickbeard logging works. In the log method it checks the log level and if it's an error or a warning it adds the msg to a list. This list is the one that's presented in the UI. 
If I move this logic to a standard logging handler and add this handler to the loggers that we want, the error/warning from subliminal/tornado or any other module that uses standard python logging will be added to that list. Then any class/module that uses standard python logging api will could have the same capabilities/features of sickbeard logger, including app error submission

**Comment by ratoaq2** [comment]
All these features/capabilities in sickbeard logger could be moved to standard log handlers and log filters,  then we could use standard logging api in our code

**Comment by fernandog** [comment]
Too much change in code?

**Comment by ratoaq2** [comment]
No... Only moving the very same logic to logging handlers and logging filters.

**Comment by ratoaq2** [comment]
Done.
- All logging messages inside subliminal and tornado follow the same logging pattern.
- All errors and warnings inside subliminal and tornado can be submitted.
- We could starting using standard python logging API instead of sickbeard Logger.

Right now, tornado and subliminal are enabled. Still... I would like to have a way to define different levels per module in the UI, but this is for a later moment.

**Comment by fernandog** [comment]
Awesome! Nice job!

File: sickbeard/logger.py [hunk 2]
```diff
@@ -64,9 +66,52 @@
     'DB': DB,
 }
 
+SSL_ERRORS = {
+    r'error \[Errno \d+\] _ssl.c:\d+: error:\d+\s*:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert internal error',
+    r'error \[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE\] sslv3 alert handshake failure \(_ssl\.c:\d+\)',
+}
+
+SSL_ERRORS_WIKI_URL = 'http://git.io/vVaIj'
```
Comment (ratoaq2) [inline sickbeard/logger.py]: Forgot to add https. I'll do it tomorrow

Comment (fernandog) [inline sickbeard/logger.py]: @ratoaq2
Should I change & squash & merge ?

**Comment by fernandog** [comment]
Approved

**Pull request merged**